src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -728,208 +751,266 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
         }
 
-        function onVolumeChange() {
-            htmlMediaHelper.saveVolume(this.volume);
-            events.trigger(self, 'volumechange');
+        /**
+         * @private
+         * @param e {Event} The event received from the `<video>` element
+         */
+        onVolumeChange = (e) => {
+            /**
+             * @type {HTMLMediaElement}
+             */
+            const elem = e.target;
+            htmlMediaHelper.saveVolume(elem.volume);
+            events.trigger(this, 'volumechange');
         }
 
-        function onNavigatedToOsd() {
-            var dlg = videoDialog;
+        /**
+         * @private
+         */
+        onNavigatedToOsd = () => {
+            const dlg = this.videoDialog;
             if (dlg) {
                 dlg.classList.remove('videoPlayerContainer-onTop');
 
-                onStartedAndNavigatedToOsd();
+                this.onStartedAndNavigatedToOsd();
             }
         }
 
-        function onStartedAndNavigatedToOsd() {
+        /**
+         * @private
+         */
+        onStartedAndNavigatedToOsd() {
             // If this causes a failure during navigation we end up in an awkward UI state
-            setCurrentTrackElement(subtitleTrackIndexToSetOnPlaying);
+            this.setCurrentTrackElement(this.subtitleTrackIndexToSetOnPlaying);
 
-            if (audioTrackIndexToSetOnPlaying != null && self.canSetAudioStreamIndex()) {
-                self.setAudioStreamIndex(audioTrackIndexToSetOnPlaying);
+            if (this.audioTrackIndexToSetOnPlaying != null && this.canSetAudioStreamIndex()) {
+                this.setAudioStreamIndex(this.audioTrackIndexToSetOnPlaying);
             }
         }
 
-        function onPlaying(e) {
-            if (!self._started) {
-                self._started = true;
-                this.removeAttribute('controls');
+        /**
+         * @private
+         * @param e {Event} The event received from the `<video>` element
+         */
+        onPlaying = (e) => {
+            /**
+             * @type {HTMLMediaElement}
+             */
+            const elem = e.target;
+            if (!this._started) {
+                this._started = true;
+                elem.removeAttribute('controls');
 
                 loading.hide();
 
-                htmlMediaHelper.seekOnPlaybackStart(self, e.target, self._currentPlayOptions.playerStartPositionTicks, function () {
-                    if (currentSubtitlesOctopus) {
-                        currentSubtitlesOctopus.timeOffset = (self._currentPlayOptions.transcodingOffsetTicks || 0) / 10000000 + currentTrackOffset;
-                        currentSubtitlesOctopus.resize();
-                        currentSubtitlesOctopus.resetRenderAheadCache(false);
+                htmlMediaHelper.seekOnPlaybackStart(this, e.target, this._currentPlayOptions.playerStartPositionTicks, function () {
+                    if (this.currentSubtitlesOctopus) {
```

Comment:
```suggestion
                    if (this.currentSubtitlesOctopus != null ) {
```

Reply:
This shouldn't make a difference, because `null` is falsy

Reply:
when i tested it, you get an error because it's undefined

Reply:
But `undefined` also evaluates to `false`, so it has to be something else

Reply:
I think it is because this is an object. I think you could do `this.hasOwnProperty('currentSubtitlesOctopus')` instead

Reply:
Can't reproduce it or did you explicitly play something with aas? (which I can't test)

Reply:
I'll check that, but I need to update my server first

Reply:
Feel free to check with/without it. When I tested it, it worked with the change and didn't without

Reply:
From what I can tell, and hopefully someone else can provide some input, it is throwing an error because `this` is undefined, therefore trying to access a property of something that is undefined is causing the error. If `this` is checked first and then the property it works 
`                    if (this && this.currentSubtitlesOctopus) {`

Reply:
That completely fixes the PR though so hopefully some other people can help check it to make sure there's not any other errors that I have missed

Reply:
That shouldn't be possible I think my server is updating right now, so we'll see

Reply:
Works for me, can't test with ass though

Reply:
Can you confirm it with the latest commits?

Reply:
I'll do it now